### ISSUE TITLE
Does not detect JSX variable usage outside of a createClass

### ISSUE THREAD

**Issue by ryanzec** [eslint/eslint#1997]
I am running into a similar issue as mentioned here:

https://github.com/babel/babel-eslint/issues/5

Now I am not using babel, just the default eslint parser.  I will have code like this:

```javascript
var React = require('react/addons');
var Desktop = require('../desktop/desktop.component.jsx');
var Router = require('react-router');
var Route = Router.Route;
var DefaultRoute = Router.DefaultRoute;

var routes = (
    <Route handler={Application}>
        <DefaultRoute handler={Desktop} />
        <Route path="/desktop" handler={Desktop} />
    </Route>
);

module.exports = routes;
```

and it will spit out these errors:

```
Route is defined but never used          no-unused-vars
DefaultRoute is defined but never used   no-unused-vars
'DefaultRoute' is not defined            no-undef
'Route' is not defined                   no-undef
```

Lucky my routes are define in one file so I can just ignore the entire file however it would be nice for this to be fixed.

**Comment by nzakas** [comment]
We're changing how we deal with JSX. See #1911 for discussion.
